scripts/cddl/generate.js

```diff
@@ -13,54 +13,55 @@ const specParsed = parse5.parse(specContent)
 /**
  * format the result and sort pre content to their corresponding classes
  */
-const cddlContent = getCDDLNodes([specParsed]).reduce((val, node) => {
-    if (!val[node.nodeClass]) {
-        val[node.nodeClass] = []
-    }
-    val[node.nodeClass].push(node.content)
-    return val
-}, {})
+const cddlNodes = getCDDLNodes([specParsed]).reduce((val, node) => {
+  if (!val[node.nodeClass]) {
+    val[node.nodeClass] = []
+  }
+  val[node.nodeClass].push(node.content)
+  return val
+}, {});
 
 /**
  * iterate over all pre cddl classes found in the document, sanitize the
  * content and write them to file.
  */
-Object.entries(cddlContent).forEach(([cddlClass, content]) => {
-    const fileContent = content
-        /**
-         * remove obsolete whitespace
-         */
-        .map((entry) => {
-            const preceedingSpace = entry.split('\n').reduce((prev, line) => {
-                if (line.length === 0) {
-                    return prev
-                }
+Object.entries(cddlNodes).forEach(([cddlName, cddlContents]) => {
+  const fileContent = cddlContents
+    /**
+     * remove obsolete whitespace
+     */
+    .map((entry) => {
+      const preceedingSpace = entry.split('\n').reduce((prev, line) => {
+        if (line.length === 0) {
+          return prev
+        }
 
-                const spaces = line.match(/^\s+/)
-                if (spaces) {
-                    return Math.min(prev, spaces[0].length)
-                }
+        const spaces = line.match(/^\s+/)
+        if (spaces) {
+          return Math.min(prev, spaces[0].length)
+        }
 
-                return 0
-            }, Infinity)
-            return entry
-                .split('\n')
-                .map((line) => line.slice(preceedingSpace))
-        })
-        /**
-         * drop all cddl snippets without declarations, e.g.
-         * ```
-         * EmptyResult
-         * ```
-         */
-        .filter((pre) => {
-            return pre.filter(Boolean).length > 1
-        })
-        /**
-         * join code lines
-         */
-        .map((pre) => pre.join('\n'))
-        .join('\n')
+        return 0
+      }, Infinity);
 
-    fs.writeFileSync(path.join(process.cwd(), `${cddlClass.split('-').shift()}.cddl`), fileContent)
-})
\ No newline at end of file
+      return entry
+        .split('\n')
+        .map((line) => line.slice(preceedingSpace))
+    })
+    /**
+     * drop all cddl snippets without declarations, e.g.
+      * ```
+      * EmptyResult
+      * ```
+      */
+    .filter((pre) => {
+      return pre.filter((line) => line.includes("=")).length > 0
```

Comment:
This change fixes the issue that single line declarations are dropped. Basically each definition should include an equal sign, so it should be easy this way to filter out types only listed to indicate what gets returned.

Reply:
I think this makes it better in the short term, but is the wrong fix. We should give the sections a different class name in the spec so that they don't get selected as part of the schema (probably just by removing the `cddl-local` and `cddl-remote` classes).